serveradmin/serverdb/query_committer.py

```diff
@@ -206,17 +206,17 @@ def _delete_servers(changed, deleted, deleted_servers):
 def _create_servers(attribute_lookup, created):
     created_servers = {}
     for attributes in created:
-        if 'hostname' not in attributes:
+        if 'hostname' not in attributes or not attributes['hostname']:
             raise CommitError('"hostname" attribute is required.')
         hostname = attributes['hostname']
 
-        if 'servertype' not in attributes:
+        if 'servertype' not in attributes or not attributes['servertype']:
             raise CommitError('"servertype" attribute is required.')
         servertype = _get_servertype(attributes)
 
-        if 'intern_ip' not in attributes:
-            raise CommitError('"intern_ip" attribute is required.')
-        intern_ip = attributes['intern_ip']
+        intern_ip = None
+        if 'intern_ip' in attributes:
+            intern_ip = attributes['intern_ip']
```

Comment:
Could be replaced with:
```suggestion
        intern_ip = attributes.get('intern_ip')
```
The `dict.get` method is giving the value for the key or `None` if the key does not exist.